pennylane/data/attributes/operator/operator.py

```diff
@@ -214,11 +222,19 @@ def _ops_to_hdf5(
             if isinstance(op, Tensor):
                 self._ops_to_hdf5(bind, op_key, op.obs)
                 op_wire_labels.append("null")
-            elif isinstance(op, qml.Hamiltonian):
+            elif isinstance(op, (qml.Hamiltonian, qml.ops.LinearCombination)):
                 coeffs, ops = op.terms()
                 ham_grp = self._ops_to_hdf5(bind, op_key, ops)
                 ham_grp["hamiltonian_coeffs"] = coeffs
                 op_wire_labels.append("null")
+            elif isinstance(op, (qml.ops.Prod, qml.ops.Sum)):
+                self._ops_to_hdf5(bind, op_key, op.operands)
```

Comment:
If an operand is `SProd`, it gets converted to `hdf5` properly? Asking because `SProd` has its own logic in the lines below.

Reply:
Yes. As the function is recursive in the subsequent recursive call with the operands, `SProd`'s logic will be executed to serialize that operand.

Reply:
Added a test for this, just in case.